Sources/Atomic.swift

```diff
@@ -55,10 +48,12 @@ internal struct UnsafeAtomicState<State: RawRepresentable> where State.RawValue
 	///   - next: The state to transition to.
 	///
 	/// - returns: `true` if the transition succeeds. `false` otherwise.
-	internal func tryTransition(from expected: State, to next: State) -> Bool {
-		return OSAtomicCompareAndSwap32Barrier(expected.rawValue,
-		                                       next.rawValue,
-		                                       value)
+	internal mutating func tryTransition(from expected: State, to next: State) -> Bool {
+		return withUnsafeMutablePointer(to: &value) { valuePointer in
+			return OSAtomicCompareAndSwap32Barrier(expected.rawValue,
+			                                       next.rawValue,
+			                                       valuePointer)
+		}
```

Comment:
I don't see this bit being dropped in the foreseeable future. It is essential, well-known part of the `inout` model since Swift 1.0.

Moreover, it is fair to say that anything concurrent written in Swift is in fact depending on "undocumented" implementation details of the Swift compiler, since Swift defines no concurrent and memory model.

😅